### PR TITLE
Master uom jem

### PR DESCRIPTION
**Pull request by jem-odoo**
Task: https://www.odoo.com/web?debug#id=1820076&view_type=form&model=project.task&action=333&active_id=965&menu_id=4720

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/uom/models/uom_uom.py [hunk 1]
```diff
@@ -54,6 +66,27 @@ def _onchange_uom_type(self):
...
+    @api.constrains('category_id', 'uom_type', 'active')
+    def _check_reference_per_category_existance(self):
+        """ Force the existence of only one UoM reference per category
+            NOTE: this is a constraint on the all table. This might not be a good practice, but this is
+            not possible to do it in SQL directly.
+        """
+        category_ids = self.mapped('category_id').ids
+        self._cr.execute("""
+            SELECT C.id AS category_id, count(U.id) AS uom_count
+            FROM uom_category C
```
Comment (tde-banana-odoo) [inline addons/uom/models/uom_uom.py]: Why C and not category ?

File: addons/uom/models/uom_uom.py [hunk 2]
```diff
@@ -54,6 +66,27 @@ def _onchange_uom_type(self):
...
+            LEFT JOIN uom_uom U ON C.id = U.category_id AND uom_type = 'reference'
```
Comment (tde-banana-odoo) [inline addons/uom/models/uom_uom.py]: Why U and not unit ?

File: addons/uom/models/uom_uom.py [hunk 3]
```diff
@@ -54,6 +66,27 @@ def _onchange_uom_type(self):
         if self.uom_type == 'reference':
             self.factor = 1
 
+    @api.constrains('category_id', 'uom_type', 'active')
+    def _check_reference_per_category_existance(self):
```
Comment (tde-banana-odoo) [inline addons/uom/models/uom_uom.py]: _check_category_reference_uniqueness ?

File: addons/uom/models/uom_uom.py [hunk 4]
```diff
@@ -2,14 +2,25 @@
...
 class UoMCategory(models.Model):
     _name = 'uom.category'
     _description = 'Product UoM Categories'
 
     name = fields.Char('Name', required=True, translate=True)
+    measure_type = fields.Selection([
+        ('unit', 'Units'),
+        ('weight', 'Weight'),
+        ('time', 'Time'),
+        ('length', 'Length / distance'),
```
Comment (tde-banana-odoo) [inline addons/uom/models/uom_uom.py]: Why not just Length ?

File: addons/uom/models/uom_uom.py (hunk 1 above)
Comment (jem-odoo) [reply addons/uom/models/uom_uom.py]: Purpose of alias in SQL is to write less. If the gain is only 4 letters with the alias, I'd rather not used it. Also, in my database bible (https://staff.info.unamur.be/dbm/Documents/Ouvrages/Dunod-2015/BD-2015-PremieresPages.pdf), many aliases with an uppercase are used. Old habits die hard ;)

File: addons/uom/models/uom_uom.py (hunk 3 above)
Comment (jem-odoo) [reply addons/uom/models/uom_uom.py]: Yeah, this looks smarter !

File: addons/uom/models/uom_uom.py (hunk 4 above)
Comment (jem-odoo) [reply addons/uom/models/uom_uom.py]: The complete name of the UoM category in odoo data is 'Length and distance'. I will change it.

File: addons/uom/models/uom_uom.py (hunk 1 above)
Comment (tde-banana-odoo) [reply addons/uom/models/uom_uom.py]: Are you paid more by stripping characters ?

File: addons/uom/models/uom_uom.py (hunk 4 above)
Comment (tde-banana-odoo) [reply addons/uom/models/uom_uom.py]: Is there any valuable difference between length and distance ?

File: addons/uom/models/uom_uom.py (hunk 3 above)
Comment (tde-banana-odoo) [reply addons/uom/models/uom_uom.py]: 3 times smarter !

File: addons/uom/models/uom_uom.py (hunk 1 above)
Comment (jem-odoo) [reply addons/uom/models/uom_uom.py]: Not sure anymore about the contract I signed some years ago. The name was "OpenERP", but it is floue ....

File: addons/uom/models/uom_uom.py (hunk 4 above)
Comment (jem-odoo) [reply addons/uom/models/uom_uom.py]: I don't think so ...

**Pull request closed without merging**